packages/next/server/node-polyfill-fetch.js

```diff
@@ -1,8 +1,22 @@
 import fetch, { Headers, Request, Response } from 'node-fetch'
+import { Agent as HttpAgent } from 'http'
+import { Agent as HttpsAgent } from 'https'
 
 // Polyfill fetch() in the Node.js environment
 if (!global.fetch) {
-  global.fetch = fetch
+  const httpAgent = new HttpAgent({ keepAlive: true })
+  const httpsAgent = new HttpsAgent({ keepAlive: true })
```

Comment:
Should we implement some other keepAlive optimizations like the ones done in this module? https://www.npmjs.com/package/agentkeepalive for example the `socket.setNoDelay` optimization seems interesting

Reply:
There are a lot of knobs to turn here.

The default [`maxSockets`](
https://nodejs.org/api/http.html#http_new_agent_options) and [`maxTotalSockets`](
https://nodejs.org/api/http.html#http_new_agent_options) are set to Infinity. So configuring these wouldn't really increase performance but might improve stability.

Using `socket.setNoDelay()` seems like it could improve performance in some cases, but the default delayed behavior (Nagles Algorithm) already "optimizes throughput at the expense of latency".

Finally, theres [`scheduling`](https://nodejs.org/api/http.html#http_new_agent_options) that was actually changed in Node.js 15.6.0 from FIFO to LIFO in https://github.com/nodejs/node/pull/36685